packages/medusa/src/services/cart.js

```diff
@@ -818,14 +819,43 @@ class CartService extends BaseService {
     return session
   }
 
-  /**
-   * Sets a payment method for a cart.
-   * @param {string} cartId - the id of the cart to add payment method to
-   * @param {PaymentMethod} paymentMethod - the method to be set to the cart
-   * @returns {Promise} result of update operation
-   */
-  async setPaymentMethod(cartId, paymentMethod) {
+  async authorizePaymentMethod(
+    cartId,
+    providerId,
+    paymentMethod,
+    context = {}
+  ) {
     const cart = await this.retrieve(cartId)
+
+    // Ensure that payment method is valid for cart
+    await this.validatePaymentMethod_(cart, paymentMethod)
+
+    const authorizedPayment = await this.paymentProviderService_.authorizePayment(
+      cart,
+      providerId,
+      paymentMethod,
+      context
+    )
+
+    paymentMethod.data = authorizedPayment
+
+    return this.cartModel_
+      .updateOne(
+        {
+          _id: cart._id,
+        },
+        {
+          $set: { payment_method: paymentMethod },
+        }
+      )
+      .then(result => {
+        // Notify subscribers
+        this.eventBus_.emit(CartService.Events.UPDATED, result)
+        return result
+      })
+  }
+
+  async validatePaymentMethod_(cart, paymentMethod) {
     const region = await this.regionService_.retrieve(cart.region_id)
```

Comment:
`region` is not used

Reply:
What do you mean? It is used two downs below your comment.